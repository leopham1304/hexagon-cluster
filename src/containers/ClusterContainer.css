.addHexagonWrapper {
  padding: 20px;
}
