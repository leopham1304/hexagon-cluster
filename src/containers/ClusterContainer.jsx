import React from "react";
import HexagonGrid from "../components/HexagonGrid";
import { Select, Input, Button, InputNumber, Row, Col } from "antd";
import {
  useClusterContainer,
  useCreateCluster,
} from "./ClusterContainer.utils";
import "./ClusterContainer.css";

const { Option } = Select;

const ClusterContainer = () => {
  const {
    hexagonList,
    onChangeRootHexagonInput, 
    onChangeTargetHexagonInput,
    onChangePositionInput,
    onAddHexagon,
    onSelectHexagon,
    onRemoveHexagon,
    setCluster, 
    rootHexagonInput,
    targetHexagonInput,
    positionInput,
    selectedHexagon,
    setHexagonList,
  } = useClusterContainer();

  useCreateCluster({ setCluster, setHexagonList });

  return (
    <div>
      <h2 className="clusterHeader">Covid Cluster</h2>
      <div>
        <Row
          gutter={[12, 12]}
          align="middle"
          justify="center"
          className="addHexagonWrapper"
        >
          <Col span={4}>
            <div id="label"> Root Hexagon </div>
            <Input
              value={rootHexagonInput}
              onChange={onChangeRootHexagonInput}
            />
          </Col>
          <Col span={4}>
            <div id="label"> Target Hexagon </div>
            <Input
              value={targetHexagonInput}
              onChange={onChangeTargetHexagonInput}
            />
          </Col>
          <Col span={1}>
            <div id="label"> Position </div>
            <InputNumber
              value={positionInput}
              onChange={onChangePositionInput}
              min={0}
              max={5}
            />
          </Col>
          <Col span={4}>
            <Button onClick={onAddHexagon}> Add Hexagon </Button>
          </Col>
        </Row>
        <Row align="middle" justify="center">
          <Col span={3}>
            <Select
              showSearch
              style={{ width: 200 }}
              placeholder="Select a hexagon"
              optionFilterProp="children"
              value={selectedHexagon}
              onChange={onSelectHexagon}
              filterOption={(input, option) =>
                option.children.toLowerCase().indexOf(input.toLowerCase()) >= 0
              }
            >
              {hexagonList.map((hex) => (
                <Option value={hex.name} key={hex.name}>
                  {hex.name}
                </Option>
              ))}
            </Select>
          </Col>
          <Col span={1}>
            <Button onClick={onRemoveHexagon}> Remove Hexagon </Button>
          </Col>
        </Row>
      </div>
      <HexagonGrid hexagonList={hexagonList} />
    </div>
  );
};

export default ClusterContainer;
