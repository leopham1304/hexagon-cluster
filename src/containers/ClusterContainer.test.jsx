import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import ClusterContainer from "./ClusterContainer.jsx";

describe("ClusterContainer", () => {
  const wrapper = shallow(<ClusterContainer />);

  it("should render the ClusterContainer Component correctly", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
