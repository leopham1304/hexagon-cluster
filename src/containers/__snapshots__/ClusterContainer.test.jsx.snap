// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClusterContainer should render the ClusterContainer Component correctly 1`] = `
<div>
  <h2
    className="clusterHeader"
  >
    Covid Cluster
  </h2>
  <div>
    <Row
      align="middle"
      className="addHexagonWrapper"
      gutter={
        Array [
          12,
          12,
        ]
      }
      justify="center"
    >
      <Col
        span={4}
      >
        <div
          id="label"
        >
           Root Hexagon 
        </div>
        <Input
          onChange={[Function]}
          type="text"
          value=""
        />
      </Col>
      <Col
        span={4}
      >
        <div
          id="label"
        >
           Target Hexagon 
        </div>
        <Input
          onChange={[Function]}
          type="text"
          value=""
        />
      </Col>
      <Col
        span={1}
      >
        <div
          id="label"
        >
           Position 
        </div>
        <ForwardRef
          max={5}
          min={0}
          onChange={[Function]}
          step={1}
          value=""
        />
      </Col>
      <Col
        span={4}
      >
        <Button
          block={false}
          ghost={false}
          htmlType="button"
          loading={false}
          onClick={[Function]}
        >
           Add Hexagon 
        </Button>
      </Col>
    </Row>
    <Row
      align="middle"
      gutter={0}
      justify="center"
    >
      <Col
        span={3}
      >
        <Select
          bordered={true}
          choiceTransitionName="zoom"
          filterOption={[Function]}
          onChange={[Function]}
          optionFilterProp="children"
          placeholder="Select a hexagon"
          showSearch={true}
          style={
            Object {
              "width": 200,
            }
          }
          transitionName="slide-up"
          value=""
        />
      </Col>
      <Col
        span={1}
      >
        <Button
          block={false}
          ghost={false}
          htmlType="button"
          loading={false}
          onClick={[Function]}
        >
           Remove Hexagon 
        </Button>
      </Col>
    </Row>
  </div>
  <HexagonGrid
    hexagonList={Array []}
  />
</div>
`;
